### ISSUE TITLE
Some policies don't apply

### ISSUE THREAD

**Issue by KoenJongUn** [flyve-mdm/glpi-plugin#912]
Hello people, my name is Koen, new here on github!

Since some time now I am trying to set up Flyvemdm plugin on GLPI under Linux.
I think it really looks promising, so well done on that part!
Actually I have come far already, I can deploy an android device, push application, some basic things.

I have off course red the documentation, but I am not a coder. Also I have completed the wizard in flyvemdm itself and followed all the steps in there. 

However now I am actually stuck. Some policies, like disable Bluetooth or camera, they work just fine. But other policies seems to do nothing at all. For example the "Disable unkown sources" is one of them. How is that possible? Are some policies still under development? Or is there something in the documentation I missed? Since disable camera, disable bluetooth, push an app, geolocation does work, it should automatically mean that the rest of my setup (LAMP + GLPI + Fusioninventory and Flyvemdm plugin + Mosquitto) should be "Oke" ... right?

I have also tried these policies on a rooted device, it made no difference at all.
Also I noticed, that for example when using adb logcat, that the mqtt message is quite long for the disable camera policy, but when applying for example the disable unknown sources, the mqtt message is very short only. Also, I opeded up the "godmode" in the agent app on the device. The test policies there do not work for me either. I don't know were it does go wrong with the policies that do not work, is it the Agent...? Is it something in GLPI? Etc etc...   

What I am looking for here, is somebody who is willing to help me a bit further. Or maybe I am in the wrong place here, I don't know!

The specs I am using is as following:

Samsung SM-T365 tablet, with android 5.1.1 Lollipop
Agent version 2.0.0 rc7 (mqtt) 

Then I have a PC running Debian 9, GLPI 9.4.3, FusionInventory 9.4 + 1.1 and Flyve mdm 2.0.0 rc-3 installed on it. 

Any help would be greatly appreciated! Thanks alot in advance!
Kind regards,
Koen

**Comment by stonebuzz** [comment]
I @KoenJongUn , 

some policies require system privileges, i. e. the Flyve MDM agent must be installed in priv_app folder (under system folder).

![image](https://user-images.githubusercontent.com/7335054/69604591-2e4bbc00-101e-11ea-9bf9-42cd36bfdc44.png)

Another example, if you want to install / uninstall apk, agent need to be isntalle as priv_app and need rooted device.
Otherwise the agent will ask to the user for a confirmation to install / uninstall apk.

another point, a lot of development has been done on the develop branch for a next release, can you do your tests on this branch?

Concerning MQTT, with the recent changes introduced by Google on the management of services in background tasks, the MQTT version works with the version 8 and 9 of android but only if the application is in foreground

For the moment we stay focused on the FCM part of the agent, while we find a workaround solution.

Best regards

**Comment by KoenJongUn** [comment]
Thank you @ stonebuzz! 

I already knew that I had to install the app as a system app, and I believe I took the right steps to do this. I can push an app silently to a device, so that kind of proves that its installed as a system app right?

I have always been using the MQTT version of the agent application, so, as I had understood its more usefull to continue with the FCM version of the agent? Do I need to change alot to switch between those 2? 

Offcourse I want to test the app and report back here! First need to switch to FCM though...

**Comment by stonebuzz** [comment]
You need to change glpi configuration to push notification through FCM rather than MQTT.

If you use android 5 you continue with MQTT version.

You should test the branch develop which has evolved a lot.

Best regards

**Comment by KoenJongUn** [comment]
Hi, I have built a develop branch apk (both version, debug and normal version. Suddeny I get the error message "org.flyve.mdm.agent.core.mqtt.mqttmodel, showdetailerror (506) - not authorized to connect. 
What could this mean? In the 2018 release I did not have this problem. I have already tried to reinstall mosquitto and checked some settings, however I cant find what this could be...

**Comment by stonebuzz** [comment]
Have you google apps or open happy installed ?

**Comment by stonebuzz** [comment]
Can and issue To the good repo, here  je glpi flyve plugin

Thanks

**Comment by KoenJongUn** [comment]
No, not that I know of. Just a standard Lolipop clear install (withch is rooted by the way) but this worked fine for the older version. I am sorry but your last message I dont understand....

**Comment by stonebuzz** [comment]
Your problem ("org.flyve.mdm.agent.core.mqtt.mqttmodel, showdetailerror (506) - not authorized to connect.) is related to the FLyveMDM Android agent and not to the FlyveMDM plugin.
You need to create new issue to this repo -> https://github.com/flyve-mdm/android-mdm-agent

Best regards
